### ISSUE TITLE
Add compile Instructions

### ISSUE THREAD

**Issue by fionera** [Aniket999969/XDP-SYNPROXY#1]
How do I compile this? I tried with clang-13 and this command:

```
clang-13 -nostdinc -O3 -g -Wall -Werror -Wno-compare-distinct-pointer-types -Wno-unused-function -I/usr/include -I/usr/include/x86_64-linux-gnu -I/usr/src/linux-headers-5.10.0-19-common/include/ -I./include -I /usr/lib/llvm-13/lib/clang/13.0.1/include synproxy-xdp.c
```

I had to disable the bpf/bpf.h and asm/unaligned.h imports because they threw too many errors

**Comment by bugsfixing** [comment]
clang -O2 -g -Wall -target bpf -c synproxy-xdp.c -o  synproxy-xdp.o

**Comment by fionera** [comment]
But with which include directories? I now get `/usr/include/linux/errno.h:1:10: fatal error: 'asm/errno.h' file not found`

This is my Compile Env:
```
docker run --rm -it -v $PWD:/host debian /bin/bash
cd /host
apt-get update && apt-get install -y clang-13 linux-headers-amd64 libbpf-dev
clang-13 -O2 -g -Wall -target bpf -c synproxy-xdp.c -o synproxy-xdp.o
```

**Comment by Tizian-Maxime-Weigt** [comment]
synproxy-xdp.c:4:10: fatal error: 'uapi/asm-generic/errno.h' file not found
#include <uapi/asm-generic/errno.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

**Comment by bugsfixing** [comment]
> synproxy-xdp.c:4:10: fatal error: 'uapi/asm-generic/errno.h' file not found
> #include <uapi/asm-generic/errno.h>
>          ^~~~~~~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> 

What OS are you using?

Please install - sudo apt install gcc-multilib
And sudo apt install linux-libc-dev

**Comment by Tizian-Maxime-Weigt** [comment]
hey,

debian 12

**Comment by Tizian-Maxime-Weigt** [comment]
root@test-01:~#  sudo apt install gcc-multilib
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
gcc-multilib ist schon die neueste Version (4:12.2.0-3).
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
root@test-01:~# sudo apt install linux-libc-dev
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
linux-libc-dev ist schon die neueste Version (6.1.119-1).
linux-libc-dev wurde als manuell installiert festgelegt.
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
root@test-01:~# ^C

**Comment by bugsfixing** [comment]
This is an OS specific error,

Please refer to https://stackoverflow.com/questions/14795608/asm-errno-h-no-such-file-or-directory

**Comment by Tizian-Maxime-Weigt** [comment]
Ahh yea i have also read this. ok so i swtich to Ubuntu.

**Comment by Tizian-Maxime-Weigt** [comment]
Update also pls your Discord.
Thanks!

**Comment by bugsfixing** [comment]
> Update also pls your Discord.
> Thanks!
> 

That would be hijack_bgp

**Comment by Tizian-Maxime-Weigt** [comment]
hijack_bgp?

**Comment by bugsfixing** [comment]
> hijack_bgp?

Yes, send me a friend request that's my discord user if I'm correct.
